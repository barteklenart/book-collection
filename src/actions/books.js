export const  addBook = (id = 1, title = '', description = '', completedAt = 0) => {
	return {
		type: 'ADD_BOOK',
		book: {
			id,
			title,
			description,
			completedAt
		}
	};
};

export const removeBook = (id) => {
	return {
		type: 'REMOVE_BOOK',
		id
	};
};

export const editBook = (id, update) => {
	return {
		type: 'EDIT_BOOK',
		id,
		update
	};
};