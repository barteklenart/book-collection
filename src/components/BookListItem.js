import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const BookListItem = ({ book, removeBook }) => {

	return (
		<div>
			<h3>{book.title}</h3>
			<p>{book.description}</p>
			<span>{moment(book.completedAt).format('MM-DD-YYYY')}</span>
			<Link to={`/edit/${book.id}`}>Edit</Link>
			<button onClick={() => removeBook(book.id)}>Remove</button>
			<hr />
		</div>
	);
};

export default BookListItem;