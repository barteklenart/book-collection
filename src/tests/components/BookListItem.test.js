import React from 'react';
import { shallow } from 'enzyme';

import BookListItem from '../../components/BookListItem';
import books from '../fixtures/book';

test('should render BookListItem correctly', () => {
	const wrapper = shallow(<BookListItem book={books[0]} />);
	expect(wrapper).toMatchSnapshot();
});

test('should handle removeBook', () => {
	const removeBook = jest.fn();
	const wrapper = shallow(<BookListItem book={books[0]} removeBook={removeBook}/>)
	expect(wrapper).toMatchSnapshot();
	wrapper.find('button').simulate('click');
	expect(removeBook).toHaveBeenCalledWith(books[0].id);
});