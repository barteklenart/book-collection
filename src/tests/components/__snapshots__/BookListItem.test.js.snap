// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle removeBook 1`] = `
<div>
  <h3>
    Hary Poter
  </h3>
  <p>
    Some description
  </p>
  <span>
    01-25-2018
  </span>
  <Link
    replace={false}
    to="/edit/1234"
  >
    Edit
  </Link>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
  <hr />
</div>
`;

exports[`should render BookListItem correctly 1`] = `
<div>
  <h3>
    Hary Poter
  </h3>
  <p>
    Some description
  </p>
  <span>
    01-25-2018
  </span>
  <Link
    replace={false}
    to="/edit/1234"
  >
    Edit
  </Link>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
  <hr />
</div>
`;
